### PR TITLE
Improve Windows Compatibility(for lib/THCUNN)

### PR DESCRIPTION
**Pull request by peterjc123**
Win64 support for lib/THCUNN

### CODE REVIEW

File: torch/lib/THCUNN/CMakeLists.txt
Comment (fmassa) [inline torch/lib/THCUNN/CMakeLists.txt]: Remove the comment?

File: torch/lib/THCUNN/generic/SpatialConvolutionLocal.cu
Comment (fmassa) [inline torch/lib/THCUNN/generic/SpatialConvolutionLocal.cu]: Was that a bug in the code?

File: torch/lib/THCUNN/generic/SpatialConvolutionMM.cu
Comment (fmassa) [inline torch/lib/THCUNN/generic/SpatialConvolutionMM.cu]: Same comment here, why was this missing?

File: torch/lib/THCUNN/generic/SpatialDepthWiseConvolution.cu
Comment (fmassa) [inline torch/lib/THCUNN/generic/SpatialDepthWiseConvolution.cu]: Same here

File: torch/lib/THCUNN/generic/SpatialDilatedConvolution.cu
Comment (fmassa) [inline torch/lib/THCUNN/generic/SpatialDilatedConvolution.cu]: Same here. Did you change the api of `col2im`?

File: torch/lib/THCUNN/generic/SpatialFractionalMaxPooling.cu
Comment (fmassa) [inline torch/lib/THCUNN/generic/SpatialFractionalMaxPooling.cu]: While I'm ok with that, does `THCTensor_(nDimension)` returns an `int`?

File: torch/lib/THCUNN/generic/SpatialFullConvolution.cu
Comment (fmassa) [inline torch/lib/THCUNN/generic/SpatialFullConvolution.cu]: Was this PR made on top of a different branch than master?

**Review by fmassa — CHANGES_REQUESTED** [review]
Thanks for the PR!
There are some things that I didn't understand. Did you start off from a different branch than master? There were a number of changes in `SpatialFullConvolution` that are not meant to be in this PR. Same with the change in function signature in `col2im`. Could you look into that?

File: torch/lib/THCUNN/generic/SpatialFullConvolution.cu
Comment (TestItOnlyOnce) [inline torch/lib/THCUNN/generic/SpatialFullConvolution.cu]: I am extremely sorry. The blame is on me for this one. The merge was done with older branch and I just copied some cu files over as I got tired and other files being different in just types.

**Review by fmassa — APPROVED** [review]
This now looks good to me, thanks!
I'm just curious though why the THCUNN build was not launched here, maybe I'm looking at the wrong place?
Once again, let's wait until the other PRs are reviewed so we merge them all together.

**Comment by peterjc123** [comment]
The build is failing because there're C wrappers for THCUNN. It checks types for it and finds out that int64_t is not one of the type supported. The C wrappers that supported int64_t is in another pull. That's the reason, I think.

**Comment by soumith** [comment]
thanks @peterjc123 !!!

**Pull request merged**
